content/en/docs/tasks/configure-pod-container/quality-service-pod.md

```diff
@@ -26,11 +26,62 @@ scheduling and evicting Pods.
 
 ## QoS classes
 
-When Kubernetes creates a Pod it assigns one of these QoS classes to the Pod:
+Quality of Services classes classify pods into one of three types: Guaranteed, Burstable, and BestEffort.
+The classification of pods is based on the [resource limits and requests](/docs/concepts/configuration/manage-resources-containers/).
+Each classification has effects on the scheduling policy and Eviction policy of a pod as well as on the resource usage of nodes.
+
+### Guaranteed
+
+To be sure a pod will be included in the Guaranteed QoS, every container in the pod must have
+resource(memory and CPU) limits and requests set, and they both have to be the same. The scheduler will make sure that
+a pod running in a Guaranteed QoS class is assigned to a node that has enough resources to fulfill its resource requests.
+A pod running in a Guaranteed QoS class can make use of exclusive CPUs  thanks to the [static](/docs/tasks/administer-cluster/cpu-management-policies/#static-policy) CPU management policy.
+
+When a node becomes [out of resource](/docs/tasks/administer-cluster/out-of-resource/), QoS classification helps kubelet
+to prioritize pods when the node needs to reclaim resources or when `oom_killer` needs to kill containers.
+If the `Guaranteed` pod uses memory resource lower than the requested resource, the pod has the lowest
```

Comment:
```
The kubelet avoids evicting `Guaranteed` Pods that are using less memory than their request.
```

I'd assume that _system-cluster-critical_ and _system-node-critical_ Pods get an even lower eviction likelihood; is that right?

Reply:
for system-cluster-critical and system-node-critical Pods it depends on priorityClass that's referenced in /docs/tasks/administer-cluster/out-of-resource/#evicting-end-user-pods and I referenced in line below. Is it needed to be mentioned again?

Reply:
That term _lowest_ is important. If there is a _system-cluster-critical_ Pod running along with a `Guaranteed` Pod and another Pod that isn't special, which of those 3 Pods is least likely to be evicted when “the node needs to reclaim resources”?

Also, is the reclaim behavior specific to the memory resource?

Reply:
```suggestion
 so if the `Guaranteed` pod uses memory resource lower than the requested resource, then the kubelet ranks pods by their [`priorityClass`](/docs/concepts/configuration/pod-priority-preemption/) and then by the percentage of requested resource usage, the lowest in the ranking 
```
because pods request memory and CPU and kubelet gets signals for disk and memory resource. yes we can say it's just memory